projects/control-service/projects/pipelines_control_service/src/integration-test/java/com/vmware/taurus/datajobs/it/DataJobListManyJobsIT.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2021 VMware, Inc.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.vmware.taurus.datajobs.it;
+
+import com.vmware.taurus.ControlplaneApplication;
+import com.vmware.taurus.datajobs.it.common.BaseIT;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+
+import static com.vmware.taurus.datajobs.it.common.WebHookServerMockExtension.*;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.user;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    classes = ControlplaneApplication.class)
+public class DataJobListManyJobsIT extends BaseIT {
+
+  private static final String DEFAULT_QUERY_WITH_VARS =
+      "query($filter: [Predicate], $search: String, $pageNumber: Int, $pageSize: Int) { "
+          + " jobs(pageNumber: $pageNumber, pageSize: $pageSize, filter: $filter, search: $search)"
+          + " {    content {      jobName      config {        team        description       "
+          + " schedule {          scheduleCron        }      }    }    totalPages    totalItems  }"
+          + "}";
+
+  private static final int JOB_COUNT = 50;
+
+  @Test
+  public void testListManyJobs() throws Exception {
+    createDummyJobs();
+
+    mockMvc
+        .perform(
+            get(String.format("/data-jobs/for-team/%s/jobs", TEST_TEAM_NAME))
+                .with(user("user"))
+                .param("query", DEFAULT_QUERY_WITH_VARS)
+                .param(
+                    "variables",
+                    "{"
+                        + "\"filter\": ["
+                        + "    {"
+                        + "      \"property\": \"config.team\","
+                        + "      \"pattern\": \""
+                        + TEST_TEAM_NAME
+                        + "\""
+                        + "    }"
+                        + "  ],"
+                        + "\"pageNumber\": 1,"
+                        + "\"pageSize\": "
```

Comment:
does the issue has anything to do with page size? what is the default page size currently, 50? is that why you create 50 jobs, to check the paging on page 2? could you then adjust the page size filter query?